packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -280,14 +271,11 @@ export class DebugConfigurationManager {
         await WaitUntilEvent.fire(this.onWillProvideDebugConfigurationEmitter, {});
     }
 
-    protected getInitialConfigurationContent(initialConfigurations: DebugConfiguration[]): string {
-        return `{
-  // Use IntelliSense to learn about possible attributes.
-  // Hover to view descriptions of existing attributes.
-  "version": "0.2.0",
-  "configurations": ${JSON.stringify(initialConfigurations, undefined, '  ').split('\n').map(line => '  ' + line).join('\n').trim()}
-}
-`;
+    protected getInitialConfigurationContent(initialConfigurations: DebugConfiguration[]): {version: string, configurations: DebugConfiguration[]} {
+        return {
```

Comment:
Can we preserve comments?

For example set `{}` as `launch`, resolve `configUri` after that and then write to this URI as before?

Reply:
now comments are preserved. but i think it makes the code more complex.